contributor-docs/versioning.md

```diff
@@ -0,0 +1,132 @@
+> 2. **MINOR** version when you add functionality in a backwards compatible
+>    manner, and
+> 3. **PATCH** version when you make backwards compatible bug fixes.
+>
+> _Additional labels for pre-release and build metadata are available as
+> extensions to the MAJOR.MINOR.PATCH format._
+
+As a result, whenever you see a `minor` or `patch` update for a package from the
+Primer you should feel confident that you can update without
+anything breaking in your project. For a full list of changes and their
+corresponding semver bumps, check out the [changes](#changes) table below.
+
+For a full list of releases, visit our [releases](https://github.com/primer/react/releases) page.
+
+## Changes
+
+| Category   | Type of change                                                | semver bump |
+| :--------- | :------------------------------------------------------------ | :---------- |
+| Component  | A component is added                                          | `minor`     |
+|            | A component is deprecated                                     | `major`     |
+|            | A component is removed                                        | `major`     |
+| Props      | A prop is added to a component                                | `minor`     |
+|            | The type of a prop is made more general                       | `minor`     |
+|            | The type of a prop is made more specific                      | `major`     |
+|            | A prop is deprecated                                          | `minor`     |
+|            | A prop is removed from a component                            | `major`     |
+|            | The element to which additional props are added to is changed | `<todo>`    |
+| Markup     | The DOM node that an `id` corresponds to is changed           | `<todo>`    |
+|            | The DOM node that an `aria-label` corresponds to is changed   | `<todo>`    |
+|            | The `role` of a component is changed                          | `<todo>`    |
+| Styles     | The `position` of the outermost element is updated            | `<todo>`    |
+|            | The `display` property of the outermost element is updated    | `<todo>`    |
+|            | A flex or grid property is updated                            | `<todo>`    |
+|            | A selector is added                                           | `<todo>`    |
+|            | A selector is removed                                         | `<todo>`    |
+|            | The specificity of a selector is raised                       | `<todo>`    |
+|            | The specificity of a selector is lowered                      | `<todo>`    |
+| Behavior   | Interactions are added to a component                         | `<todo>`    |
+|            | Interactions are removed from a component                     | `<todo>`    |
+| TypeScript | A type is added                                               | `minor`     |
```

Comment:
Do you have any examples of this one in mind? I think this could also be a `patch` in certain scenarios?

Reply:
Really just was thinking of a narrow case where a type has been added to the Public API of the package.

In this scenario, I framed it as a `minor` since it would be additive and would require someone to explicitly import it in order to use it.

Let me know if there is a better way to talk about types in this kind of context, I am honestly not sure how to frame type changes in terms of semver so was just applying a `propTypes` mindset to this kind of section 👀

Reply:
ah gotcha, thanks for clarifying. I think a type addition that doesn't make a material API change, or something that doesn't require the user to react to can go into a patch. A prominent new feature however, like a new component entirely - which includes new types - would constitute a minor in my mind. I agree that it's nuanced however. Perhaps we can split this up into different type edge cases.